usb/device/dev_cdc/main.c

```diff
@@ -0,0 +1,80 @@
+/**
+ * Copyright (c) 2024 Raspberry Pi (Trading) Ltd.
+ *
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+
+#include <stdlib.h>
+#include <bsp/board_api.h>
+#include <tusb.h>
+
+#include <pico/stdio.h>
+
+void custom_cdc_task(void);
+
+int main(void)
+{
+    // Initialize TinyUSB stack
+    board_init();
+    tusb_init();
+
+    // TinyUSB board init callback after init
+    if (board_init_after_tusb) {
+        board_init_after_tusb();
+    }
+
+    // let pico sdk use the first cdc interface for std io
+    stdio_init_all();
+
+    // main run loop
+    while (1) {
+        // TinyUSB device task | must be called regurlarly
+        tud_task();
+        
+        // custom tasks
+        custom_cdc_task();
+    }
+
+    // indicate no error
+    return 0;
+}
+
+void custom_cdc_task(void)
+{
+    // polling CDC interfaces if wanted
+
+    // Check if CDC interface 0 (for pico sdk stdio) is connected and ready
+
+    if (tud_cdc_n_connected(0)) {
+        // print on CDC 0 some debug message
+        printf("Connected to CDC 0\n");
+        sleep_ms(5000); // wait for 5 seconds
```

Comment:
Given that `custom_cdc_task` is called from within the main loop, does this `sleep` here mean that once CDC 0 is connected, `tud_task` will only get called once every 5 seconds? :thinking:  (and doesn't `tud_task` need to get called more often than that in order to keep the USB connection alive?)